ui/store/actions.ts

```diff
@@ -3550,6 +3550,18 @@ export function removePermissionsFor(
   };
 }
 
+export function updateNetworksList(
+  networksList: [],
+): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
+  return (dispatch) => {
+    callBackgroundMethod('updateNetworksList', [networksList], (err) => {
+      if (err) {
+        dispatch(displayWarning(err));
```

Comment:
This `displayWarning` method is deprecated, we should avoid using it for anything. This `callBackgroundMethod` method is deprecated instead.

I'd suggest using `submitRequestToBackground` instead, and letting the error bubble up instead of catching it